templates/template.go

```diff
@@ -8,32 +8,77 @@ import (
 	"io"
 	"text/template"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/gobuffalo/packr"
 )
 
-// NewTemplate will create a temp file with the template for a CFN stack
-func NewTemplate(assetName string, data interface{}) (io.Reader, error) {
+// GetAsset reads an asset from "disk" into a string
+func GetAsset(assetName string, options ...func(string, *string) error) (string, error) {
 	box := packr.NewBox("./assets")
 
 	asset, err := box.MustString(assetName)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
+	assetRef := aws.String(asset)
 
-	tmpl, err := template.New(assetName).Parse(string(asset[:]))
```

Comment:
It's not always necessary. I found only a few places where GetAsset (was NewTemplate) actually needed to parse the template. For those methods I explicitly chained AddData as one of the functional arguments. The majority of references to GetAsset are for policies now. 

I struggled with whether to leave in the template parsing or taking it out. In the end I pulled it out because a) it's not always necessary b) the method signature is smaller c) and the method only does what you want it to do by default (get an asset).

Let me know if you still feel strongly about refactoring this. I can roll AddData back into GetAsset.

Reply:
I misunderstood...what you have is the correct implementation.  All the code related to parsing golang templates makes sense to be in a single method.  I still struggle with the name `AddData`...i think i'm missing something.  let me read it a few more times :)